iptables/iptables.go

```diff
@@ -98,7 +99,17 @@ func New() (*IPTables, error) {
 func NewWithProtocol(proto Protocol) (*IPTables, error) {
 	path, err := exec.LookPath(getIptablesCommand(proto))
 	if err != nil {
-		return nil, err
+		// Add /usr/sbin and /sbin to $PATH
+		env := os.Getenv("PATH")
+		if env != "" {
+			env = env + ":"
+		}
+		os.Setenv("PATH", env+"/usr/sbin:/sbin")
```

Comment:
I have multiple concerns over this:
 * I do believe this has process-wide effects, thus influencing at a distance other parts of an application using this library
 * in case of multiple calls with failures, it will keep polluting the environment over and over
 * it seems a bad behavior for a library to manipulate the environment behind the back of the user and the application

Reply:
Agreed - libraries should never manipulate $PATH.

If you'd like to be able to call non-system iptables, then I would accept another New function that took an explicit path to the ip(6)tables binary. It needs to validate that the command matches the protocol.

Reply:
Ugh, I totally agree with you, it's my fault. Actually I've realized the change has some process-wide side effects, I then took a look at the os package, confirmed it's thread-safe, and forgot it somewhere out of my brain. Give me a few minutes, let me fix it.